File: 2주차(250331)/이유안/BOJ_2002.py
'''
대근: 들어가는 순서
영식: 나오는 순서

추월이 문제

'''
from collections import deque
n = int(input())
in_order = deque([input() for _ in range(n)])
out_order = deque([input() for _ in range(n)])

cnt = 0
while out_order:
    outcar = out_order.popleft()
    waiting = deque([])
    check = False
    while in_order:
        incar = in_order.popleft()
        if incar == outcar:
            break
        else:
            check = True
            waiting.append(incar)
    waiting.extend(in_order)
    in_order = waiting
    if check:
        cnt+=1
print(cnt)

File: 2503/2주차(250331)/이유안/BOJ_2002.java
import java.util.*;
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int res = 0;
        Deque<String> inOrder = new ArrayDeque<>();
        Deque<String> outOrder = new ArrayDeque<>();

        for (int i=0; i<n; i++){
            inOrder.offer(br.readLine());
        }
        for (int i=0; i<n; i++){
            outOrder.offer(br.readLine());
        }

        while (!outOrder.isEmpty()){
            String now = outOrder.poll();
            Deque<String> waiting = new ArrayDeque<>();
            boolean check = false;
            while(!inOrder.isEmpty()){
                String inCar = inOrder.poll();
                if (inCar.equals(now)){
                    break;
                } else{
                    check = true;
                    waiting.add(inCar);
                }
            }
            // inOrder 갱신해주기
            while (!inOrder.isEmpty()){
                waiting.add(inOrder.poll());
            }
            inOrder = waiting;

            if (check) res++;
        }

        System.out.println(res);

    }
}